ui/workspaceMonitor.js

```diff
@@ -41,28 +40,17 @@ class WorkspaceMonitor extends GObject.Object {
         if (!this._enabled)
             return;
 
-        function _isLastWindow(apps, win) {
-            if (apps.length === 0)
-                return true;
-
-            if (apps.length > 1)
-                return false;
-
-            const windows = apps[0].get_windows();
-            return windows.length === 1 && windows[0] === win;
-        }
-
-        const visibleApps = this._getVisibleApps();
-        if (_isLastWindow(visibleApps, actor.meta_window))
+        const windows = this._getVisibleWindows();
+        if (windows.length === 1)
```

Comment:
I guess there will always be 1 window at least which is the actor disappearing here? Or should this check if len==0?